controller/token.go

```diff
@@ -250,11 +252,14 @@ func (c *TokenController) Retrieve(ctx *app.RetrieveTokenContext) error {
 		return jsonapi.JSONErrorResponse(ctx, err)
 	}
 	if externalToken != nil {
-		updatedToken, err := c.updateProfileIfEmpty(ctx, providerConfig, externalToken)
-		if err != nil {
-			return jsonapi.JSONErrorResponse(ctx, err)
+		if externalToken.Username == "" || (ctx.ForcePull != nil && *ctx.ForcePull == true) {
```

Comment:
1. `if *ctx.ForcePull == true {...}` looks weird :) `if *ctx.ForcePull {...}` would look better ;)
2. Let's move the check of this param to the `updateProfileIfEmpty(ctx, config, token, force)` method?